src/events/ready.test.ts

```diff
@@ -0,0 +1,123 @@
+import { deployCommands } from '../helpers/actions/deployCommands';
+const mockDeployCommands = deployCommands as jest.Mock;
+
+// Mock revokeCommands so we can track it
+jest.mock('../helpers/actions/revokeCommands');
+import { revokeCommands } from '../helpers/actions/revokeCommands';
+const mockRevokeCommands = revokeCommands as jest.Mock;
+
+// Mock verifyCommandDeployments so we can track it
+jest.mock('../helpers/actions/verifyCommandDeployments');
+import { verifyCommandDeployments } from '../helpers/actions/verifyCommandDeployments';
+const mockVerifyCommandDeployments = verifyCommandDeployments as jest.Mock;
+
+// Mock the logger so the code doesn't print to the console
+jest.mock('../logger');
+import { getLogger } from '../logger';
+const mockGetLogger = getLogger as jest.Mock;
+mockGetLogger.mockImplementation(() => {
+	return {
+		info: () => undefined,
+	} as unknown as Console;
+});
+
+// Import the code to test
+import { ready } from './ready';
+
+describe('once(ready)', () => {
+	beforeEach(() => {
+		// Default is no deploy, no revoke, no method behavior
+		mockParseArgs.mockReturnValue({
+			deploy: false,
+			revoke: false,
+		});
+		mockDeployCommands.mockResolvedValue(undefined);
+		mockRevokeCommands.mockResolvedValue(undefined);
+		mockSetActivity.mockReturnValue({});
+	});
+
+	afterEach(() => {
+		jest.restoreAllMocks();
+	});
+
+	test("doesn't touch commands if the `deploy` and `revoke` flags are not set", async () => {
+		mockParseArgs.mockReturnValue({
+			deploy: false,
+			revoke: false,
+		});
+		await expect(ready.execute(client)).resolves.toBeUndefined();
+		expect(mockDeployCommands).not.toHaveBeenCalled();
+		expect(mockRevokeCommands).not.toHaveBeenCalled();
+	});
+
+	test('deploys commands if the `deploy` flag is set', async () => {
+		mockParseArgs.mockReturnValue({
+			deploy: true,
+			revoke: false,
+		});
+		await expect(ready.execute(client)).resolves.toBeUndefined();
+		expect(mockDeployCommands).toHaveBeenCalledWith(new MockClient());
+		expect(mockRevokeCommands).not.toHaveBeenCalled();
+	});
+
+	test('revokes commands if the `revoke` flag is set', async () => {
+		mockParseArgs.mockReturnValue({
+			deploy: false,
+			revoke: true,
+		});
+		await expect(ready.execute(client)).resolves.toBeUndefined();
+		expect(mockDeployCommands).not.toHaveBeenCalled();
+		expect(mockRevokeCommands).toHaveBeenCalledWith(new MockClient());
+	});
+
+	test('deploys commands if both the `revoke` and `deploy` flags are set', async () => {
+		mockParseArgs.mockReturnValue({
+			deploy: true,
+			revoke: true,
+		});
+		await expect(ready.execute(client)).resolves.toBeUndefined();
+		expect(mockDeployCommands).toHaveBeenCalledWith(new MockClient());
+		expect(mockRevokeCommands).not.toHaveBeenCalled();
+	});
+
+	test('verifies command deployments', async () => {
+		await expect(ready.execute(client)).resolves.toBeUndefined();
+		expect(mockVerifyCommandDeployments).toHaveBeenCalledWith(new MockClient());
```

Comment:
Shouldn't these reference the same client instance?
```suggestion
		expect(mockVerifyCommandDeployments).toHaveBeenCalledWith(client);
```

This might work anyway as-is because Jest is comparing the properties of the objects, rather than using a strict identity comparison, but that's an implementation detail that might change, based on the verbiage of the assertion method. (IMO, "have been called with" implies strict identity in the case of non-scalar values.)

Reply:
Oh you're right, I meant to make that change after figuring out how to mock only the Client, but I forgot a few spots apparently.